examples_tests/47.DerivMapTest/main.cpp

```diff
@@ -228,9 +228,12 @@ static core::smart_refctd_ptr<asset::ICPUImage> createDerivMapFromHeightMap(asse
 	state.axisWraps[1] = _vwrap;
 	state.axisWraps[2] = asset::ISampler::ETC_CLAMP_TO_EDGE;
 	state.borderColor = _borderColor;
+	state.enableLUTUsage = true;
 	state.scratchMemoryByteSize = DerivativeMapFilter::getRequiredScratchByteSize(&state);
 	state.scratchMemory = reinterpret_cast<uint8_t*>(_NBL_ALIGNED_MALLOC(state.scratchMemoryByteSize, _NBL_SIMD_ALIGNMENT));
 
+	state.computePhaseSupportLUT(&state);
```

Comment:
should return a bool whether it succeeded, also do some validation of the state

Reply:
Almost all validation is already done by `validate`.
I can think of one more: `phaseCount[axis] == 0` despite `axis` being a valid/to-be-filtered axis, so I'm putting that in.
What else you'd like me to validate?

Reply:
it depends how much you'll split out into `IBlitUtilities`

Reply:
Not much, as far as I can see right now, it will one be `E_ALPHA_SEMANTIC` and `getPhaseCount`. We could also put `computePhaseSupportLUT` there but its implementation depends on the kernels (template params of `CBlitImageFilter`) along with weird kernel scaling things happening there. If we keep `computePhaseSupportLUT` in `CBlitImageFilter` we can reuse/rely on its `validate` method.

Reply:
Yes if you want to compute the LUT as a standalone, `computePhaseSupportLUT` not only needs to know the 3 kernels but also the scaling factor (ratio of output resolution to input resolution), otherwise its impossible to recalculate the kernels supports and evaluate them.

_Kernels get "stretched" so that 1.0 unit in kernel function definition corresponds to 1 pixel distance in output image, thats why there's this "weird" stuff going on._

P.S. I thought the main reason to create `IBlitUtilities` was to move LUT precomputation there?

Reply:
> P.S. I thought the main reason to create IBlitUtilities was to move LUT precomputation there?

I only wanted it to have the trivial things, which were otherwise locked behind a template instantiation of `CBlitImageFilter`, like `E_ALPHA_SEMANTIC` and `getPhaseCount`.
Is there a benefit of doing the weird scaling things in the utilities class?

Reply:
you can work out the support which you need for the compute scanner's smem calculation I guess ?

Reply:
when you asked about refactoring out a common-base, you mentioned about the weirdness of having to build a CPU Blit state to fire off GPU work

Anyway if you want a non-templated base you can have `IBlitUtilities` and then a templated `CBlitUtilities`

Reply:
> Anyway if you want a non-templated base you can have IBlitUtilities and then a templated CBlitUtilities

You're right, I want this.